src/AggregateFunctions/AggregateFunctionMap.cpp

```diff
@@ -0,0 +1,133 @@
+                    return std::make_shared<AggregateFunctionMap<Int64>>(nested_function, arguments);
+                case TypeIndex::UInt8:
+                    return std::make_shared<AggregateFunctionMap<UInt8>>(nested_function, arguments);
+                case TypeIndex::Date:
+                case TypeIndex::UInt16:
+                    return std::make_shared<AggregateFunctionMap<UInt16>>(nested_function, arguments);
+                case TypeIndex::DateTime:
+                case TypeIndex::UInt32:
+                    return std::make_shared<AggregateFunctionMap<UInt32>>(nested_function, arguments);
+                case TypeIndex::UInt64:
+                    return std::make_shared<AggregateFunctionMap<UInt64>>(nested_function, arguments);
+                case TypeIndex::UUID:
+                    return std::make_shared<AggregateFunctionMap<UInt128>>(nested_function, arguments);
+                case TypeIndex::FixedString:
+                case TypeIndex::String:
+                    return std::make_shared<AggregateFunctionMap<String>>(nested_function, arguments);
+                default:
+                    throw Exception{"Illegal columns in arguments for combinator " + getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+            }
+        }
+        else if (!arguments.empty())
+        {
+            // check if we got tuple of arrays or just arrays and if so, try to redirect to sum/min/max-MappedArrays to implement old behavior
+            auto nested_func_name = nested_function->getName();
+            if (nested_func_name == "sum" || nested_func_name == "min" || nested_func_name == "max")
+            {
+                bool match;
+                const auto * tup_type = checkAndGetDataType<DataTypeTuple>(arguments[0].get());
+
+                auto check_func = [](DataTypePtr t)
+                {
+                    return t->getTypeId() == TypeIndex::Array;
+                };
+
+                if (tup_type)
+                {
+                    const auto & types = tup_type->getElements();
+                    match = arguments.size() == 1 && types.size() >= 2 && std::all_of(types.begin(), types.end(), check_func);
+                }
+                else
+                {
+                    // sumMappedArrays and others support more than 2 mapped arrays
+                    match = arguments.size() >= 2 && std::all_of(arguments.begin(), arguments.end(), check_func);
+                }
+
+                if (match)
+                {
+                    AggregateFunctionProperties out_properties;
+                    auto & aggr_func_factory = AggregateFunctionFactory::instance();
+                    return aggr_func_factory.get(nested_func_name + "MappedArrays", arguments, params, out_properties);
+                }
+            }
+        }
+
+        throw Exception{"Illegal columns in arguments for combinator " + getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
```

Comment:
This exception message is too general. It would be better to concrete what is wrong with arguments (wrong type, wrong number of arguments, etc)